tests/golos.vesting_tests.cpp

```diff
@@ -244,35 +317,58 @@ BOOST_FIXTURE_TEST_CASE(delegate_vesting, golos_vesting_tester) try {
     BOOST_CHECK_EQUAL(err.delegation_no_funds2, vest.delegate_vesting(N(sania), N(vania), vest.make_asset(min_remainder+1)));
     BOOST_TEST_MESSAGE("--- succeed when withdraval counted");
     BOOST_CHECK_EQUAL(success(), vest.delegate_vesting(N(sania), N(vania), amount));
-    // BOOST_CHECK_EQUAL(success(), vest.delegate_vesting(N(sania), N(issuer), amount));    // TODO: fix: sending to issuer instead of vania fails
-    // TODO: check delegation to account without balance
-    // TODO: check change delegation
-    // TODO: check min step
-    // TODO: check min remainder
-    // TODO: check delegation object
-    // TODO: check battery limitation
+//    // BOOST_CHECK_EQUAL(success(), vest.delegate_vesting(N(sania), N(issuer), amount));    // TODO: fix: sending to issuer instead of vania fails
+//    // TODO: check delegation to account without balance
+//    // TODO: check change delegation
+//    // TODO: check min step
+//    // TODO: check min remainder
+//    // TODO: check delegation object
+//    // TODO: check battery limitation
 } FC_LOG_AND_RETHROW()
 
 BOOST_FIXTURE_TEST_CASE(undelegate_vesting, golos_vesting_tester) try {
     BOOST_TEST_MESSAGE("Test un-delegating vesting shares");
     prepare_balances();
 
     // undelegation looks broken due composite index used in smart-contract
+    auto vesting_withdraw = vest.vesting_withdraw(13, 120);
+    auto vesting_amount = vest.vesting_amount(10 * 1e3);
+    auto delegation = vest.delegation(5e6, 15e6, 0, 0, 120);
+
+    auto issuer = N(issuer);
+    auto params = "[" + vesting_withdraw + "," + vesting_amount + "," + delegation + "]";
+    BOOST_CHECK_EQUAL(success(), vest.set_params(issuer, _vesting_sym, params));
+    auto t = vest.get_params(_vesting_sym);
+    BOOST_TEST_MESSAGE("--- " + fc::json::to_string(t));
+    produce_block();
+
+    BOOST_CHECK_EQUAL(true, t.is_object());
 
+    auto obj = mvo(t);
     const int divider = vest.make_asset(1).get_amount();
-    const int min_step = cfg::delegation.min_amount / divider;
-    const int min_remainder = cfg::delegation.min_remainder / divider;
+    const int min_step = obj["delegation_params"]["min_amount"].as<uint64_t>() / divider;
```

Comment:
actually, you don't need to get values of parameters from db to use in checks. it's better use constants you passed to `init_params`, so if something was wrong while setting parameters, we'll catch this if expected (calculated from constants) values are different from actual (from db).

the only test, where you need to check db values is `set_params` test